### ISSUE TITLE
Unable to send messages to servers that don't support 8-bit encoding

### ISSUE THREAD

**Issue by njeyaakili** [thunderbird/thunderbird-android#1206]
### Expected behaviour

Message should be delivered to outgoing mail server
### Actual behaviour

Get a "failed to send message" error, message is put in drafts
### Steps to reproduce
1. create message
2. send
### Environment

K-9 Mail version: 5.108

Android version: 6.0.1
Account type (IMAP, POP3, WebDAV/Exchange): imap (but this is only a send issue, retrieving mail works fine).
[fail-to-send.txt](https://github.com/k9mail/k-9/files/186083/fail-to-send.txt)

This is against Rackspace's mail service -- secure.emailsrvr.com -- port 465 - ssl/tls
They only support TLS1.0 and above. 
Picking up mail from their server works fine, as does the outgoing mail server test during setup/configuration. Did an upgrade to the 5.108 alpha on this device, and a re-install on samsung android 5.x devices, encountering the same issue on all. The same K-9 works fine sending to gmail, with basically the same outgoing mail server settings (465 - ssl/tls).

**Comment by Valodim** [comment]
```
java.lang.RuntimeException: nope
at com.fsck.k9.mailstore.BinaryMemoryBody.setEncoding(BinaryMemoryBody.java:36)
at com.fsck.k9.mail.internet.MimeMessage.setEncoding(MimeMessage.java:462)
at com.fsck.k9.mail.internet.MimeMessage.setUsing7bitTransport(MimeMessage.java:701)
at com.fsck.k9.mail.transport.SmtpTransport.sendMessageTo(SmtpTransport.java:499)
at com.fsck.k9.mail.transport.SmtpTransport.sendMessage(SmtpTransport.java:488)
at com.fsck.k9.controller.MessagingController.sendPendingMessagesSynchronous(MessagingController.java:3113)
at com.fsck.k9.controller.MessagingController$20.run(MessagingController.java:2993)
at com.fsck.k9.controller.MessagingController.run(MessagingController.java:257)
at java.lang.Thread.run(Thread.java:818)
```

looks like BinaryMemoryBody doesn't support setEncoding which is indirectly called by SmtpTransport

**Comment by njeyaakili** [comment]
Thanks. I'm trying to figure out why this happens with sending to the rackspace server but not gmail's. I have basically the same settings -- including send only text, no html, etc., and the gmail ones work fine. In short, I'm trying to see if there's a work-around as I'd rather not have to downgrade to .107.

**Comment by Valodim** [comment]
The rackspace server doesn't support 8bit transfer encoding

**Comment by cketti** [comment]
We probably want to fix this by getting rid of creating 8-bit messages. See #1034

**Comment by cdmackay** [comment]
Confirmed seeing this with a large Beehive company server.

Is there a workaround to continue using 5.108 with this server, please? It's a bit of a disaster for me... thanks.

**Comment by Valodim** [comment]
Should be fixed in #1554

**Comment by gunterkoenigsmann** [comment]
Am 29. August 2016 11:34:05 MESZ, schrieb Vincent Breitmoser <notifications@github.com>:
>Should be fixed in #1554

I can acknowledge that it is and am sending this answer using the new k9 version. Thanks a lot!

And kind regards,

   Gunter.
-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.
